waddrmgr/db.go

```diff
@@ -1268,8 +1308,41 @@ func upgradeManager(namespace walletdb.Namespace) error {
 
 	// Upgrade the manager as needed.
 	if version < LatestMgrVersion {
-		// No upgrades yet.
+		// Upgrade addresses used flag
+		upgradeVersion1to2(namespace)
 	}
 
 	return nil
 }
+
+// upgradeVersion1to2 upgrades the database from version 1 to version 2
+// 'usedAddrBucketName' a bucket for storing addrs flagged as marked is initialized
+// and it will be updated on the next rescan
+func upgradeVersion1to2(namespace walletdb.Namespace) error {
+	err := namespace.Update(func(tx walletdb.Tx) error {
+		rootBucket := tx.RootBucket()
+		mainBucket := tx.RootBucket().Bucket(mainBucketName)
+
+		_, err := rootBucket.CreateBucketIfNotExists(usedAddrBucketName)
+		if err != nil {
+			str := "failed to create used addresses bucket"
+			return managerError(ErrDatabase, str, err)
+		}
+
+		var version uint32
+		var buf [4]byte
+		version = LatestMgrVersion
```

Comment:
This is wrong.  The function is explicitly to update version 1 to 2.  If you use the constant here, when it gets bumped to version 3, it would mess up the version 1 -> 2 -> 3 upgrade paths.